### ISSUE TITLE
TFEstimatorSpec cannot support gradient clipping (and other customized train_ops)

### ISSUE THREAD

**Issue by Bob-Chou** [intel-analytics/analytics-zoo#1281]
Currently `TFEstimatorSpec` can only support construct from `loss` when training.  This might cause trouble when migrating some cases with **customized training ops**, like **gradient clipping** used in [chines_ocr](https://github.com/YCG09/chinese_ocr/blob/master/ctpn/lib/fast_rcnn/train.py#L106-L111) as follows:
```python
tvars = tf.trainable_variables()
grads, norm = tf.clip_by_global_norm(tf.gradients(total_loss, tvars), 10.0)
train_op = opt.apply_gradients(list(zip(grads, tvars))
sess.run(train_op)
```
Maybe we might support passing `train_op` into `TFEstimatorSpec` just like [tf.estimator.EstimatorSpec](https://www.tensorflow.org/api_docs/python/tf/estimator/EstimatorSpec) ? Or we might at least support **gradient clipping** so that migration of `chinese_ocr` could be correct.

**Comment by hkvision** [comment]
Training BERT also need gradient clipping, we need to think of a proper API to expose, if not exactly the same as TF. @jason-dai @yangw1234

**Comment by yangw1234** [comment]
`estimator.setConstantGradientClipping`
`estimator.setGradientClippingByL2Norm`
just like the scala Estimator?

**Comment by jason-dai** [comment]
Sounds good to me

**Comment by hkvision** [comment]
OK. Then you propose a PR or I add this into my BERT PR? @yangw1234
